### PR TITLE
Add documentation for device type option

### PR DESCRIPTION
**Pull request by felipediel**
## Proposed change

Add documentation for device type option.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/32523
- This PR fixes or closes issue: https://github.com/home-assistant/core/issues/30215

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by Quentame** [comment]
Before submitting a PR, please push from an updated branch and verify your commits.

Closing

**Comment by frenck** [comment]
No need to close it @Quentame...

@felipediel Please rebase your PR onto the `next` branch to clean it up.

**Comment by felipediel** [comment]
@frenck Thank you!

What a mess I made here, excuse me for that. Yesterday I was tired and a bit overwhelmed and I screwed up on git. Let me know if you want me to open a clean PR.

**Comment by frenck** [comment]
You don't have to open a new PR, git rebase this one onto our `next` branch.

**Comment by felipediel** [comment]
I rebased and force pushed but I don't know if it worked as expected because of a merge that I had made before by mistake. This is a mess, I think I should close this PR and open a new one to avoid trouble.

**Comment by frenck** [comment]
`git rebase --onto origin/next patch-1~2`
(since only the last two commits taking into account when rebasing)

**Comment by felipediel** [comment]
Great, it worked. Thank you!

File: source/_integrations/broadlink.markdown [hunk 1]
```diff
@@ -33,6 +33,7 @@ remote:
   - platform: broadlink
     host: IP_ADDRESS
     mac: MAC_ADDRESS
+    type: 0x5f36
```
Comment (frenck) [inline source/_integrations/broadlink.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

**Review by frenck — CHANGES_REQUESTED** [review]
Left a small comment.

Furthermore, how would a user, that reads this page, known which type to use?

File: source/_integrations/broadlink.markdown (hunk 1 above)
Comment (felipediel) [reply source/_integrations/broadlink.markdown]: Done.

**Comment by frenck** [comment]
One question still remains: 

> Furthermore, how would a user, that reads this page, known which type to use?

How can a user find the `0x` value? Which are available?

**Comment by felipediel** [comment]
@frenck [These](https://github.com/mjg59/python-broadlink/blob/5daf12c86f8c046adcfde6151fcd4747eb4f882b/broadlink/__init__.py#L18-L61) are the types we know so far, but they get out of date very quickly, so I don't think it's interesting to document them.

The user can find the type of his device using the official Broadlink App. But I do not recommend this, because this application is evil. It connects the device to the cloud and prevents it from being controlled by Home Assistant. So the user should stay away from it.

The alternatives we have today are Broadlink Manager and this [debug tool](https://drive.google.com/open?id=1Gr-Qe_sdVL-Yn2g2ITDFYB1TTpd1pX8Z) I've created to debug the API we are using. For now it's a pretty shitty tool, so I think we should stick to Broadlink Manager in the docs.

It is important to note that none of this would be necessary if we implemented HELLO_REQUEST (broadlink.discover) within our integration. The device knows its type and sends it in response. It would be so much easier to just say hello. We are violating etiquette and making the user pay for it.

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by Quentame** [comment]
Sorry for the immediate closing @felipediel but you will see later on that’s it’s annoying being requested for an unchecked code.

To prevent that, some advices :
- always **rebase**, not merge (in general)
- rebase just before creating a PR/MR
- if you doubt you can create a draft PR
- check the number of files impacted (and content)

And I don’t know if you have the right but PR can be reopened, or you can create a new one, so no worries 😉 

Thanks a lot for your contribution 👍

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by felipediel** [comment]
@Quentame Yes, sorry about that. I was on a hurry late at night to send the docs and I worked on an old branch by mistake. I was so tired that I didn't even check my commits and the rest you know.
At the end of the day, that mistake was worth it, as I ended up learning how to do a decent rebase.
Thanks :thumbsup:

**Review by frenck — DISMISSED** [review]

**Review by frenck — APPROVED** [review]
Nice! Thanks, @felipediel! 👍

**Pull request merged**